### ISSUE TITLE
Reset auth when client send Authorization header.

### ISSUE THREAD

**Issue by frenche** [modauthgssapi/mod_auth_gssapi#22]
Or as Simo put it: "Forcibly reset credentials on client request".

Currently, with IIS, when the client sends a new Authorization header on an
already authenticated connection (either NTLM or when Persistent-Auth
is set to true) then the server restart the authentication and the
client will be served according to the new one.
However, with mod_auth_gssapi if the connection is authenticated the
server ignores it and serves the client by previous auth.

Some client implementations treat negotiate as purely request-based
auth and might therefore mix requests from different identities over
the same connection.
So it appears IIS behavior is more graceful in this case and maybe
worth adapting.

Something like:

```patch
--- a/src/mod_auth_gssapi.c
+++ b/src/mod_auth_gssapi.c
@@ -301,7 +301,9 @@ static int mag_auth(request_rec *req)
         mag_check_session(req, cfg, &mc);
     }

-    if (mc) {
+    auth_header = apr_table_get(req->headers_in, "Authorization");
+
+    if (mc && !auth_header) {
         /* register the context in the memory pool, so it can be freed
          * when the connection/request is terminated */
         apr_pool_userdata_set(mc, "mag_conn_ptr",
@@ -321,7 +323,6 @@ static int mag_auth(request_rec *req)
         pctx = &ctx;
     }

-    auth_header = apr_table_get(req->headers_in, "Authorization");
     if (!auth_header) goto done;

     auth_header_type = ap_getword_white(req->pool, &auth_header);
```

**Comment by frenche** [comment]
Thanks a lot!

**Comment by frenche** [comment]
Hi Simo,

On Sun, Jun 14, 2015 at 2:02 AM, simo5 <notifications@github.com> wrote:

> Closed #22 <https://github.com/modauthgssapi/mod_auth_gssapi/issues/22>
> via 5c3cc59
> <https://github.com/modauthgssapi/mod_auth_gssapi/commit/5c3cc59740925e6ba1c3447237964d07d7f96aa1>
> .
>

I've run further tests and found some issues which I have addressed here:
https://github.com/frenche/mod_auth_gssapi/commits/ba_fix

Regards,
Isaac B.

**Comment by frenche** [comment]
Note.

On Mon, Jun 15, 2015 at 12:14 AM, Isaac Boukris <iboukris@gmail.com> wrote:
> Hi Simo,
>
> On Sun, Jun 14, 2015 at 2:02 AM, simo5 <notifications@github.com> wrote:
>>
>> Closed #22 via 5c3cc59.
>
>
> I've run further tests and found some issues which I have addressed here:
> https://github.com/frenche/mod_auth_gssapi/commits/ba_fix
>
> Regards,
> Isaac B.

I think the last commit isn't good, it should only call
mag_conn_destroy and not unregister the cleanup.
I'll give it more thoughts later this week, unless you fix it in a better way :)

Thanks!

**Comment by frenche** [comment]
This old issue doesn't seem the right place.
I've opened PR #38 so you could review and merge what's right.

Thanks and regards.
